rt/src/test/java/io/fabric8/maven/rt/SpringbootHttpBoosterIT.java

```diff
@@ -0,0 +1,220 @@
+                String podOfApplication = pod.getMetadata().getLabels().get("app");
+
+                if (action.equals(Action.ADDED) && bApplicationPod) {
+                    if (KubernetesHelper.isPodReady(pod) && podOfApplication.equals(TESTSUITE_REPOSITORY_ARTIFACT_ID)) {
+                        String podStatus = KubernetesHelper.getPodStatusText(pod);
+                        applicationPod = pod;
+                        terminateLatch.countDown();
+                    }
+                }
+            }
+
+            @Override
+            public void onClose(KubernetesClientException e) { }
+        });
+
+        // Wait till pod starts up
+        while (terminateLatch.getCount() > 0) {
+            try {
+                terminateLatch.await();
+            } catch (InterruptedException aException) {
+                // ignore
+            }
+            if (applicationPod != null) {
+                break;
+            }
+        }
+        assertApplicationPodRoute(applicationPod);
+    }
+
+
+    /**
+     * Appends some annotation properties to the fmp's configuration in test repository's pom
+     * just to distinguish whether the application is re-deployed or not.
+     *
+     * @param testRepository
+     * @throws Exception
+     */
+    public void addRedeploymentAnnotations(Repository testRepository, String annotationKey, String annotationValue) throws Exception {
+        File pomFile = new File(testRepository.getWorkTree().getAbsolutePath(), "/pom.xml");
+        Model model = readPomModelFromFile(pomFile);
+
+        Xpp3Dom configurationDom = Xpp3DomBuilder.build(
+                new ByteArrayInputStream(
+                        ("<configuration>" +
+                                "<resources>" +
+                                "   <labels> " +
+                                "       <all> " +
+                                "           <property>" +
+                                "               <name>" + annotationKey + "</name> " +
+                                "               <value>" + annotationValue + "</value> " +
+                                "           </property> " +
+                                "       </all> " +
+                                "   </labels> " +
+                                "   <annotations> " +
+                                "       <all> " +
+                                "           <property>" +
+                                "               <name>" + annotationKey + "</name> " +
+                                "               <value>" + annotationValue + "</value> " +
+                                "           </property> " +
+                                "       </all> " +
+                                "   </annotations> " +
+                                "</resources>" +
+                                "</configuration>").getBytes()),
+                "UTF-8");
```

Comment:
Line 149-151, can these snippets be loaded for test classpath resources, as that will be easier to maintain and read